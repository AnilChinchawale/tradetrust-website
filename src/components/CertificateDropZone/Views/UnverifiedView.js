import PropTypes from "prop-types";
import Link from "next/link";
import { get } from "lodash";
import { TYPES, MESSAGES } from "../../../constants/VerificationErrorMessages";
import css from "./viewerStyles.scss";

const DetailedErrors = ({ verificationStatus }) => {
  const errors = [];
  if (!get(verificationStatus, "hash.checksumMatch")) errors.push(TYPES.HASH);
  if (!get(verificationStatus, "issued.issuedOnAll")) errors.push(TYPES.ISSUED);
  if (get(verificationStatus, "revoked.revokedOnAny", true))
    errors.push(TYPES.REVOKED);
  if (!get(verificationStatus, "identity.identifiedOnAll"))
    errors.push(TYPES.IDENTITY);
  const renderedError = errors.map((errorType, index) => (
    <div id={`error-tab${index}`} key={index}>
      <p className={css.messages}>{MESSAGES[errorType].failureTitle}</p>
      <p>{MESSAGES[errorType].failureMessage}</p>
    </div>
  ));
  return <div className={css.verifications}>{renderedError}</div>;
};

DetailedErrors.propTypes = {
  verificationStatus: PropTypes.object
};

const View = ({ resetData, verificationStatus }) => (
  <div
    className={`${css["viewer-container"]} ${css.invalid}`}
    style={{
      backgroundColor: "#fbeae9",
      borderRadius: 10
    }}
  >
    <span className={css["message-container"]}>
      <img src="/static/images/dropzone/invalid.svg" />
      <span className="invalid m-3" style={{ fontSize: "1.5rem" }}>
        This certificate is not valid
      </span>
    </span>

    <DetailedErrors verificationStatus={verificationStatus} />

    <Link href="/faq">
      <div className={css["unverified-btn"]}>What should I do?</div>
    </Link>

    <div className={css["secondary-links"]}>
      <span>
        <Link href=" ">
          <a
            onClick={e => {
              e.preventDefault();
              resetData();
            }}
            className={css["text-link"]}
          >
            Try another
          </a>
        </Link>
      </span>
    </div>
  </div>
);

View.propTypes = {
  handleRenderOverwrite: PropTypes.func,
  resetData: PropTypes.func,
  document: PropTypes.object,
  verificationStatus: PropTypes.object
};

export default View;
