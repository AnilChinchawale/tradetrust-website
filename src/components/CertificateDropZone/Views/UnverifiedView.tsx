import React, { FunctionComponent } from "react";
import { Link } from "react-router-dom";
import { get } from "lodash";
import { MESSAGES, TYPES } from "../../../constants/VerificationErrorMessages";
import css from "./viewerStyles.scss";

interface DetailedErrorsProps {
  verificationStatus: any; // TODO type me :)
}
const DetailedErrors: FunctionComponent<DetailedErrorsProps> = ({ verificationStatus }) => {
  const errors = [];
  if (!get(verificationStatus, "hash.checksumMatch")) errors.push(TYPES.HASH);
  if (!get(verificationStatus, "issued.issuedOnAll")) errors.push(TYPES.ISSUED);
  if (get(verificationStatus, "revoked.revokedOnAny", true)) errors.push(TYPES.REVOKED);
  if (!get(verificationStatus, "identity.identifiedOnAll")) errors.push(TYPES.IDENTITY);
  const renderedError = errors.map((errorType, index) => (
    <div key={index}>
      <p className={css.messages}>{MESSAGES[errorType].failureTitle}</p>
      <p>{MESSAGES[errorType].failureMessage}</p>
    </div>
  ));
  return (
    <div id="error-tab" className={css.verifications}>
      {renderedError}
    </div>
  );
};

interface UnverifiedViewProps {
  resetData: () => void;
  verificationStatus: any;
}
export const UnverifiedView: FunctionComponent<UnverifiedViewProps> = ({ resetData, verificationStatus }) => (
  <div
    className={`${css["viewer-container"]} ${css.invalid}`}
    style={{
      backgroundColor: "#fbeae9",
      borderRadius: 10
    }}
  >
    <span className={css["message-container"]}>
      <img src="/static/images/dropzone/invalid.svg" />
      <span className="invalid m-3" style={{ fontSize: "1.5rem" }}>
        This certificate is not valid
      </span>
    </span>

    <DetailedErrors verificationStatus={verificationStatus} />

    <div className={css["unverified-btn-container"]}>
      <Link to="/faq">
        <span className={css["unverified-btn"]}>What should I do?</span>
      </Link>
    </div>

    <div className={css["secondary-links"]}>
      <span>
        <a
          onClick={e => {
            e.preventDefault();
            resetData();
          }}
          className={css["text-link"]}
        >
          Try another
        </a>
      </span>
    </div>
  </div>
);
