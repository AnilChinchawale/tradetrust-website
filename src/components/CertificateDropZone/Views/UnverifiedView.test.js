import React from "react";
import renderer from "react-test-renderer";

import UnverifiedView from "./UnverifiedView";

const VALID_VERIFICATION_STATUS = {
  hash: {
    checksumMatch: true
  },
  issued: {
    issuedOnAll: true,
    details: [
      {
        address: "0x48399Fb88bcD031C556F53e93F690EEC07963Af3",
        issued: true
      }
    ]
  },
  revoked: {
    revokedOnAny: false,
    details: [
      {
        address: "0x48399Fb88bcD031C556F53e93F690EEC07963Af3",
        revoked: false
      }
    ]
  },
  valid: true,
  identity: {
    identifiedOnAll: true,
    details: [
      {
        identified: true,
        dns: "tradetrust.io",
        smartContract: "0x48399Fb88bcD031C556F53e93F690EEC07963Af3"
      }
    ]
  }
};

it("renders correctly when the hash mismatch", () => {
  const tree = renderer
    .create(
      <UnverifiedView
        verificationStatus={{
          ...VALID_VERIFICATION_STATUS,
          hash: { checksumMatch: false }
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when the certificate is not issued", () => {
  const tree = renderer
    .create(
      <UnverifiedView
        verificationStatus={{
          ...VALID_VERIFICATION_STATUS,
          issued: { issuedOnAll: false }
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when the certificate is revoked", () => {
  const tree = renderer
    .create(
      <UnverifiedView
        verificationStatus={{
          ...VALID_VERIFICATION_STATUS,
          revoked: { revokedOnAny: true }
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when the issuers are not identified", () => {
  const tree = renderer
    .create(
      <UnverifiedView
        verificationStatus={{
          ...VALID_VERIFICATION_STATUS,
          identity: { identifiedOnAll: false }
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
