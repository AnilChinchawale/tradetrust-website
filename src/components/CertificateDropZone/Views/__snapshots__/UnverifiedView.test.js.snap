// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when the certificate is not issued 1`] = `
<div
  className="undefined undefined"
  style={
    Object {
      "backgroundColor": "#fbeae9",
      "borderRadius": 10,
    }
  }
>
  <span>
    <img
      src="/static/images/dropzone/invalid.svg"
    />
    <span
      className="invalid m-3"
      style={
        Object {
          "fontSize": "1.5rem",
        }
      }
    >
      This certificate is not valid
    </span>
  </span>
  <div>
    <div>
      <p>
        Certificate not issued
      </p>
      <p>
        This certificate cannot be found. Please contact your issuing institution for help or issue the certificate before trying again.
      </p>
    </div>
  </div>
  <div
    onClick={[Function]}
  >
    What should I do?
  </div>
  <div>
    <span>
      <a
        href=" "
        onClick={[Function]}
      >
        Try another
      </a>
    </span>
  </div>
</div>
`;

exports[`renders correctly when the certificate is revoked 1`] = `
<div
  className="undefined undefined"
  style={
    Object {
      "backgroundColor": "#fbeae9",
      "borderRadius": 10,
    }
  }
>
  <span>
    <img
      src="/static/images/dropzone/invalid.svg"
    />
    <span
      className="invalid m-3"
      style={
        Object {
          "fontSize": "1.5rem",
        }
      }
    >
      This certificate is not valid
    </span>
  </span>
  <div>
    <div>
      <p>
        Certificate revoked
      </p>
      <p>
        This certificate has been revoked by your issuing institution. Please contact your issuing institution for more details.
      </p>
    </div>
  </div>
  <div
    onClick={[Function]}
  >
    What should I do?
  </div>
  <div>
    <span>
      <a
        href=" "
        onClick={[Function]}
      >
        Try another
      </a>
    </span>
  </div>
</div>
`;

exports[`renders correctly when the hash mismatch 1`] = `
<div
  className="undefined undefined"
  style={
    Object {
      "backgroundColor": "#fbeae9",
      "borderRadius": 10,
    }
  }
>
  <span>
    <img
      src="/static/images/dropzone/invalid.svg"
    />
    <span
      className="invalid m-3"
      style={
        Object {
          "fontSize": "1.5rem",
        }
      }
    >
      This certificate is not valid
    </span>
  </span>
  <div>
    <div>
      <p>
        Certificate has been tampered with
      </p>
      <p>
        The contents of this certificate are inaccurate and have been tampered with.
      </p>
    </div>
  </div>
  <div
    onClick={[Function]}
  >
    What should I do?
  </div>
  <div>
    <span>
      <a
        href=" "
        onClick={[Function]}
      >
        Try another
      </a>
    </span>
  </div>
</div>
`;

exports[`renders correctly when the issuers are not identified 1`] = `
<div
  className="undefined undefined"
  style={
    Object {
      "backgroundColor": "#fbeae9",
      "borderRadius": 10,
    }
  }
>
  <span>
    <img
      src="/static/images/dropzone/invalid.svg"
    />
    <span
      className="invalid m-3"
      style={
        Object {
          "fontSize": "1.5rem",
        }
      }
    >
      This certificate is not valid
    </span>
  </span>
  <div>
    <div>
      <p>
        Certificate issuer identity is invalid
      </p>
      <p>
        This certificate was issued by an invalid issuer.
      </p>
    </div>
  </div>
  <div
    onClick={[Function]}
  >
    What should I do?
  </div>
  <div>
    <span>
      <a
        href=" "
        onClick={[Function]}
      >
        Try another
      </a>
    </span>
  </div>
</div>
`;
